cli/npm/resolvers/local.rs

```diff
@@ -365,18 +364,16 @@ async fn sync_resolution_with_fs(
     }
   }
 
-  // 4. Create all the packages in the node_modules folder, which are symlinks.
+  // 4. Create all the top level packages in the node_modules folder, which are symlinks.
   //
   // Symlink node_modules/<package_name> to
   // node_modules/.deno/<package_id>/node_modules/<package_name>
   let mut found_names = HashSet::new();
-  let mut pending_packages = VecDeque::new();
-  pending_packages.extend(snapshot.top_level_packages().map(|id| (id, true)));
-  while let Some((id, is_top_level)) = pending_packages.pop_front() {
+  let mut ids = snapshot.top_level_packages().collect::<Vec<_>>();
+  ids.sort_by(|a, b| b.cmp(a)); // create determinism and only include the latest version
+  for id in ids {
     let root_folder_name = if found_names.insert(id.nv.name.clone()) {
       id.nv.name.clone()
-    } else if is_top_level {
-      id.nv.to_string()
```

Comment:
This was previously useless because it would create this non-deterministically. I don't think there's any use for this so I removed it.